src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -37,63 +40,51 @@ public class UpdateNuGetConfigPackageSourcesMappings : Task
         /// </summary>
         public string[] SourceBuildSources { get; set; }
 
-        public string VmrRoot { get; set; }
+        [Required]
+        public string SbrpRepoSrcPath { get; set; }
 
-        private string SBRPCacheSourceName = "source-build-reference-package-cache";
+        private const string SbrpCacheSourceName = "source-build-reference-package-cache";
 
         public override bool Execute()
         {
             string xml = File.ReadAllText(NuGetConfigFile);
             string newLineChars = FileUtilities.DetectNewLineChars(xml);
             XDocument document = XDocument.Parse(xml);
-            XElement pkgSrcMappingElement = document.Root.Descendants().FirstOrDefault(e => e.Name == "packageSourceMapping");
             XElement pkgSourcesElement = document.Root.Descendants().FirstOrDefault(e => e.Name == "packageSources");
             if (pkgSourcesElement == null)
             {
+                Log.LogMessage(MessageImportance.Low, "Package sources are missing.");
+
                 return true;
             }
 
+            XElement pkgSrcMappingElement = document.Root.Descendants().FirstOrDefault(e => e.Name == "packageSourceMapping");
             if (pkgSrcMappingElement == null)
             {
                 pkgSrcMappingElement = new XElement("packageSourceMapping");
                 document.Root.Add(pkgSrcMappingElement);
             }
 
-            Hashtable allSourcesPackages = new Hashtable();
-            Hashtable currentPackages = new Hashtable();
-            Hashtable referencePackages = new Hashtable();
-            Hashtable previouslyBuiltPackages = new Hashtable();
-            Hashtable oldSourceMappingPatterns = new Hashtable();
+            var allSourcesPackages = new Dictionary<string, List<string>>();
```

Comment:
Would it make sense to promote these dictionaries to instance members since they are passed around frequently?